erica_app/erica/request_processing/eric_mapper.py

```diff
@@ -0,0 +1,194 @@
+            values.get('person_a_has_merkzeichen_ag')
+        ]
+        if values.get('person_a_requests_fahrkostenpauschale'):
+            if any(merkzeichen_for_higher_fahrkostenpauschale):
+                values['person_a_fahrkostenpauschale_has_merkzeichen_bl_tbl_h_ag_pflegegrad'] = True
+            elif values.get('person_a_disability_degree') >= 80 or (values.get('person_a_has_merkzeichen_g') and values.get('person_a_disability_degree') >= 70):
+                values['person_a_fahrkostenpauschale_has_merkzeichen_g_and_degree_70_degree_80'] = True
+        return values
+
+    @root_validator(pre=True)
+    def set_person_b_merkzeichen_bl_tbl_h_pflegegrad(cls,values):
+        merkzeichen_values = [
+           values.get('person_b_has_pflegegrad'),
+           values.get('person_b_has_merkzeichen_bl'),
+           values.get('person_b_has_merkzeichen_tbl'),
+           values.get('person_b_has_merkzeichen_h')
+        ]
+        if any(merkzeichen_values) and values.get('person_b_requests_pauschbetrag'):
+            values['person_b_pauschbetrag_has_merkzeichen_bl_tbl_h_pflegegrad'] = True
+        return values
+
+    @root_validator(pre=True)
+    def set_person_b_merkzeichen_g_ag(cls, values):
+        merkzeichen_values = [
+           values.get('person_b_has_merkzeichen_g'),
+           values.get('person_b_has_merkzeichen_ag')
+        ]
+        if any(merkzeichen_values) and values.get('person_b_requests_pauschbetrag'):
+            values['person_b_pauschbetrag_has_merkzeichen_g_ag'] = True
+        return values
+
+    @root_validator(pre=True)
+    def set_person_b_fahrkostenpauschale(cls, values):
+        merkzeichen_for_higher_fahrkostenpauschale = [
+            values.get('person_b_has_pflegegrad'),
+            values.get('person_b_has_merkzeichen_bl'),
+            values.get('person_b_has_merkzeichen_tbl'),
+            values.get('person_b_has_merkzeichen_tbl'),
+            values.get('person_b_has_merkzeichen_h'),
+            values.get('person_b_has_merkzeichen_ag')
+        ]
+        if values.get('person_b_requests_fahrkostenpauschale'):
+            if any(merkzeichen_for_higher_fahrkostenpauschale):
+                values['person_b_fahrkostenpauschale_has_merkzeichen_bl_tbl_h_ag_pflegegrad'] = True
+            elif values.get('person_b_disability_degree') >= 80 or (values.get('person_b_has_merkzeichen_g') and values.get('person_b_disability_degree') >= 70):
+                values['person_b_fahrkostenpauschale_has_merkzeichen_g_and_degree_70_degree_80'] = True
+        return values
+
+    @validator('person_a_dob', 'person_b_dob', 'familienstand_date', pre=True)
+    def convert_datetime_to_d_m_y(cls, v):
+        return v.strftime(STANDARD_DATE_FORMAT) if v else None
+
+    @validator('person_a_pauschbetrag_disability_degree', 'person_b_pauschbetrag_disability_degree')
+    def set_disability_degree_only_if_above_20(cls, v):
+        return None if v and v < 20 else v
```

Comment:
The only thing I don't find so handsome is that we abuse the validation context to perform a mapping... If you have not seen this yet, you might think that the model is validated here.

Reply:
I like the fact, that you moved it from erica_input to a new class :grin:

Reply:
As far as I understood it pydantic is mainly used for parsing:

> [pydantic is primarily a parsing library, not a validation library. Validation is a means to an end: building a model which conforms to the types and constraints provided.
> In other words, pydantic guarantees the types and constraints of the output model, not the input data.
> This might sound like an esoteric distinction, but it is not. If you're unsure what this means or how it might affect your usage you should read the section about Data Conversion below.
> Although validation is not the main purpose of pydantic, you can use this library for custom validation.](https://pydantic-docs.helpmanual.io/usage/models/)

Therefore, I think this would be an ok use case. Furthermore, this should in anyway be only a step in the direction of moving the mapping into it's own set of function/classes that map an object directly to xml output for Erica.

However, I understand your concern and if you see a better way doing that, let's discuss it.